internal/cmd/flags.go

```diff
@@ -42,7 +43,14 @@ var (
 		Usage: "signature envelope format, options: 'jws', 'cose'",
 	}
 	SetPflagSignatureFormat = func(fs *pflag.FlagSet, p *string) {
-		fs.StringVar(p, PflagEnvelopeType.Name, envelope.JWS, PflagEnvelopeType.Usage)
+		defaultEnvelopeFormat := envelope.JWS
+		// load config to get envelopeType
+		config, err := configutil.LoadConfigOnce()
```

Comment:
You mean different plugin may have different envelopeType support. So we need to chose the default envelopeType based on the plugin's capability? 

I don't know what is the benefit to move it to notation-go. It is a CLI logic. Notation-go doesn't know what is CLI input. It just know it reveived an envelopeType. If the plugin doesn't not support the provided envelopeType, return an error when signing occurred.

Reply:
@JeyJeyGao

I agree that throwing an error from the plugin makes sense if a user expects an unsupported envelope format from the plugin. We are aligned on that and I closed #325.

Now, coming to where we should keep the logic that determines the envelope type based on the three sources
1. User explicitly choosing a format
2. Configured option in config.json
3. Default option of JWS when above two options are not provided.

I think there is value in having this logic in notation-go so that irrespective of the user interface (notation CLI or whatever), notation-go can select the right option based on the situation. If we go the other way (i.e. having this logic in notation CLI), then every consumer of notation-go needs to read the config file and make a decision of choosing the right envelope type and then invoke notation-go. We should keep notation CLI to be a thin wrapper around notation-go so that consumers of notation-go don't need to re-implement whatever we are implementing in notation.

Reply:
In fact, all config.json arguments are for CLI now. Notation-go provide the ability to sign and verify for different envelope types. The user of notation-go may not be the CLI. It could be another service, such as [Ratify](https://github.com/deislabs/ratify). And the user may only need one of the supported envelopeType and don't want to use the logic to chose the default one. So the library provides the multiple envelopeType ability, and the user decides how to use it, which is more reasonable.

It is true that notation CLI is a thin wrapper, but that doesn't mean every logic should be move to notation-go. The envelopeType of config.json is a helper for notation-cli flag --envelope-type to simplify the command, so the notation-go doesn't need to know the logic and only need to receive an envelopeType to start signing.

Reply:
Sounds good. Can we please update the flag from `envelope-type` to `signature-format` as suggested in https://github.com/notaryproject/notaryproject/pull/194?

Reply:
I suggest changing to `envelope-format`. It will be more consistent with the term `envelope-format` we used in the [signature-specification.md](https://github.com/notaryproject/notaryproject/blob/main/signature-specification.md#signature-envelope).

Reply:
@rgnote Please review this PR to rename the field in config.json: https://github.com/notaryproject/notation-go/pull/175/

Reply:
@FeynmanZhou 

The targeted users of Notation CLI may not know what is an envelope. `SignatureFormat` is more reasonable for a dummy user. The signature spec is targeted for pro users.

Reply:
@rgnote Make sense. I accept that we can choose `signature-format` to make dummy users easy to understand.

Reply:
@rgnote Updated https://github.com/notaryproject/notation-go/pull/175
Please review it again.

Reply:
Approved. Please update this PR and I will review again. Thanks for making this change @JeyJeyGao

Reply:
@rgnote Updated now, please review it again.